### ISSUE TITLE
Failed to bind after upgrading to java 21

### ISSUE THREAD

**Issue by pbthorste** [micronaut-projects/micronaut-micrometer#679]
### Expected Behavior

The application should have started without logging a warning

### Actual Behaviour

After application was configured to use java 21 - at startup, the application logs this warning:

```
level=WARN  : Failed to bind as java.util.concurrent.ThreadPerTaskExecutor is unsupported. thrd=[main] logger=io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics
```

Not sure if this is a micronaut or a micrometer issue. 

### Environment Information

Operating system: Linux
JDK: 21 (coretto)

### Version

4.2.2

**Comment by jeremyg484** [comment]
It looks like Micrometer does not provide any instrumentation for an `ExecutorService` that uses virtual threads. See https://github.com/micrometer-metrics/micrometer/issues/3650 and https://github.com/micrometer-metrics/micrometer/issues/3956.

I think the appropriate thing for us to do is detect usage of virtual threads in our `ExecutorServiceMetricsBinder` and not try to bind `ExecutorServiceMetrics` in that case.

Given that, I believe this particular warning can be safely ignored in the meantime.

**Comment by pbthorste** [comment]
Thank you @jeremyg484  :)

**Comment by mdeso** [comment]
I still get this warning with Micronaut platform 4.3.7, micronaut-micrometer 5.4.0

**Comment by jeremyg484** [comment]
> I still get this warning with Micronaut platform 4.3.7, micronaut-micrometer 5.4.0

It did not end up getting included in the 4.3.x releases as we've not done an updated release of this module yet. It should definitely get included in 4.4.0.
